src/object/obj_class.c

```diff
@@ -482,3 +496,64 @@ obj_ec_codec_get(daos_oclass_id_t oc_id)
 
 	return NULL;
 }
+
+
+/* Encode full stripe from SGL */
+int
+obj_encode_full_stripe(daos_obj_id_t oid, daos_sg_list_t *sgl, uint32_t *j,
+		       size_t *k, struct obj_ec_parity *parity, int p_idx)
```

Comment:
some API comments (for example for the parameters, input or output) might be helpful, thanks.

Just FYI that using "sg_idx" "sg_off" to replace j, k seems clearer (and we probably can keep the k/p using same as isa-l AFAP).